packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -80,12 +155,11 @@ export class PreferenceServiceImpl implements PreferenceService, FrontendApplica
     protected readonly providerProvider: PreferenceProviderProvider;
 
     protected readonly providers: PreferenceProvider[] = [];
+    protected providersMap: Map<PreferenceScope, PreferenceProvider>;
 
     @postConstruct()
     protected init(): void {
         this.toDispose.push(Disposable.create(() => this._ready.reject()));
-        this.providers.push(this.schema);
-        this.preferences = this.parsePreferences();
```

Comment:
Before it was ensured that one can query default values without awaiting on `ready`. Some clients are relying on it.

Reply:
please correct me if i am wrong: before my change, clients can query default values if and only if the default schema is constructed. 

with my change, i did `defaultSchema.ready.resolve()` in the constructor (https://github.com/theia-ide/theia/pull/3247/files/7c201eacd53dfa16c259013e3ac53523f2f36af7#diff-e901dd93040742d0ffb3d98186e1043eR42). So i think that behaviour is unchanged.

Reply:
Clients can query any values immediately after the preference service is constructed. The preference service should guarantee that clients always get proper default values. It is not a case anymore because the code was moved from synchronous `postContruct` callback into async `initialize` application callback.

Reply:
got it. i will move the code back to `postContruct` to make sure the prefs can be queried once the preference-service is constructed.